arcaflow_plugin_rtla/rtla_plugin.py

```diff
@@ -0,0 +1,155 @@
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE,
+                text=True,
+            )
+
+            # Block here, waiting on the cancel signal
+            self.exit.wait(params.duration)
+
+        except subprocess.CalledProcessError as err:
+            return "error", ErrorOutput(
+                f"{err.cmd[0]} failed with return code {err.returncode}:\n{err.output}"
+            )
+
+        # Secondary block interrupt is via the KeyboardInterrupt exception.
+        # This enables running the plugin stand-alone without a workflow.
+        except (KeyboardInterrupt, SystemExit):
+            print("\nReceived keyboard interrupt; Stopping data collection.\n")
+            self.finished_early = True
+
+        # In either the case of a keyboard interrupt or a cancel signal, we need to
+        # send the SIGINT to the subprocess.
+        if self.finished_early:
+            proc.send_signal(2)
+
+        output, _ = proc.communicate()
+
+        latency_hist = []
+        total_irq_latency = {}
+        total_thr_latency = {}
+        total_usr_latency = {}
+        stats_names = ["over:", "count:", "min:", "avg:", "max:"]
+        stats_per_col = []
+        found_all = False
+
+        for line in output.splitlines():
+            if re.match(r"^Index", line):
+                print("Found Index")
+                cols = line.lower().split()
+            if (re.match(r"^\d", line)) or (
+                line.split()[0] in stats_names and not found_all
+            ):
+                row_obj = {}
+                for i, col in enumerate(cols):
+                    row_obj[col] = line.split()[i]
+                if re.match(r"^\d", line):
+                    latency_hist.append(row_obj)
+                else:
+                    stats_per_col.append(row_obj)
+            if re.match(r"^ALL", line) and not found_all:
+                found_all = True
+            if found_all and line.split()[0] in stats_names:
+                if line.split()[0] != "over:":
+                    total_irq_latency[line.split()[0][:-1]] = line.split()[1]
+                    total_thr_latency[line.split()[0][:-1]] = line.split()[2]
+                    if params.user_threads:
+                        total_usr_latency[line.split()[0][:-1]] = line.split()[3]
+            else:
+                continue
+
+        if params.user_threads:
+            return "success", TimerlatOutput(
+                latency_hist,
+                stats_per_col,
+                latency_stats_schema.unserialize(total_irq_latency),
+                latency_stats_schema.unserialize(total_thr_latency),
+                latency_stats_schema.unserialize(total_usr_latency),
+            )
+
+        return "success", TimerlatOutput(
+            latency_hist,
+            stats_per_col,
+            latency_stats_schema.unserialize(total_irq_latency),
+            latency_stats_schema.unserialize(total_thr_latency),
+        )
```

Comment:
It may make sense to combine these into the same output, and just use the Ternary Operator on the value for total_usr_latency.